phpdotnet/phd/Package/Generic/XHTML.php

```diff
@@ -832,7 +833,7 @@ public function format_container_chunk_below($open, $name, $attrs, $props) {
 
         $toc = '<ol>';
         $desc = "";
-        if (!in_array($id, $this->TOC_WRITTEN)) {
+        if (!in_array($id, $this->TOC_WRITTEN ?? [])) {
```

Comment:
I was going to suggest
```suggestion
        if ($this->TOC_WRITTEN && !in_array($id, $this->TOC_WRITTEN)) {
```

But I'm guessing `TOC_WRITTEN` may be undef?

Reply:
There are some dynamic properties in some of these files...

Reply:
Ah right :|